pkg/vtpm/src/server.cpp

```diff
@@ -56,6 +114,21 @@ int readMessage(int sock, google::protobuf::uint32 size)
     CodedStrmInput.PopLimit(msgLimit);
 
     cout << "Received command is " << request.command() << std::endl;
+    if (!isCommandAllowed(request.command())) {
+        cout << "Not a legal command, bailing out" << std::endl;
+        response.set_response("Command forbidden!");
+        sendResponse(sock, response);
+        return 0;
+    }
+    //Clear all the files which we expect to be output by the cmd.
+    //This is to clear residual files of previous invocations, which
+    //otherwise may be interpreted as output of current invocation.
+    for (int i=0; i < request.expectedfiles_size(); i++) {
+        std::string expectedFile = request.expectedfiles(i);
```

Comment:
Don't you need to sanitize or sandbox the expectedfiles to prevent e.g., a domU passing 
/etc/passwd as the expected filename?

Reply:
Also, what prevents expectedFile from being arbitrary shell syntax including appending semicolon plus some bad command? Using exec instead of system would address that I think.  But I don't know the state of the art for sanitizing things like this.

Reply:
Fixed it with
a) execve() instead of system()
b) running server inside a dedicated folder
c) not allowing paths in filename